R/convert2metid.R

```diff
@@ -13,343 +13,216 @@
 #' @export
 
 convert2metid <-
-  function(data,
-           path = ".",
-           threads = 5,
-           Submitter = "Xiaotao Shen") {
-    dir.create(path, showWarnings = FALSE, recursive = TRUE)
-    message("Extracting MS1 inforamtion...")
-    all_names <-
-      data %>%
-      purrr::map(function(x) {
-        x$info$key
-      }) %>%
-      unlist() %>%
-      unique() %>%
-      sort()
-
-    progresser <-
-      show_progresser(index = seq_along(data),
-                      progresser = c(1, seq(10, 100, 10)))
-
-    ms1_info <-
-      seq_along(data) %>%
-      purrr::map(function(i) {
-        # cat(i, " ")
-        if (i %in% progresser$idx) {
-          message(progresser$progresser[which(i == progresser$idx)], " ",
-                  appendLF = FALSE)
-        }
-        x <- data[[i]]
-        x <-
-          x$info %>%
-          dplyr::arrange(key)
-
-        if (sum(duplicated(x$key)) == 0) {
-          x <-
-            t(x) %>%
-            as.data.frame()
-          colnames(x) <- as.character(x[1,])
-          x <- x[-1, , drop = FALSE]
-          new_name <-
-            setdiff(all_names, colnames(x))
-          if (length(new_name) > 0) {
-            new_x <-
-              matrix(NA, nrow = 1, ncol = length(new_name)) %>%
-              as.data.frame()
-            colnames(new_x) <- new_name
-            x <-
-              cbind(x, new_x) %>%
-              as.data.frame()
-            x <- x[, all_names]
-          }
-          return(x)
-        }
-
-        x <-
-          x %>%
-          plyr::dlply(.variables = .(key)) %>%
-          lapply(function(y) {
-            if (nrow(y) == 1) {
-              return(y)
-            }
-            y$value <-
-              paste(y$value, collapse = "{}")
-            return(y[1, , drop = FALSE])
-          }) %>%
-          dplyr::bind_rows() %>%
-          as.data.frame() %>%
-          dplyr::arrange(key)
-
-        x <-
-          t(x) %>%
+  function (data, 
+            file_names, 
```

Comment:
what are check_names and row_names here? I didn't see them in the function. Again, do you mean "raw_names"?

Reply:
pull request demo, will update right code